cpp/oneapi/dal/backend/primitives/objective_function/test/fixture.hpp

```diff
@@ -519,20 +585,61 @@ class logloss_test : public te::float_algo_fixture<Param> {
             auto out_vector_host = out_vector.to_host(this->get_queue());
             const std::int64_t st = fit_intercept ? 0 : 1;
 
+            // We use double for gth computations to achieve better precision
             for (std::int64_t i = st; i < p + 1; ++i) {
-                float_t correct = 0;
+                double correct = 0;
                 for (std::int64_t j = st; j < p + 1; ++j) {
-                    correct += vec_host.at(j - st) * hessian_host.at(i, j);
+                    correct += (double)vec_host.at(j - st) * (double)hessian_host.at(i, j);
```

Comment:
It should be dicouraged to use C-like casts in modern C++ code. Please switch to `static_caast` instead